numba/cuda/tests/cudadrv/test_cuda_driver.py

```diff
@@ -113,6 +113,21 @@ def test_cuda_driver_stream(self):
         for i, v in enumerate(array):
             self.assertEqual(i, v)
 
+    def test_cuda_driver_default_stream(self):
+        # Test properties of the default stream
+        ds = self.context.get_default_stream()
+        self.assertIn("Default CUDA stream", repr(ds))
+        self.assertEqual(0, int(ds))
+        self.assertTrue(ds)
```

Comment:
Note: `bool(Stream)` is the check that is done in memcpy to decide if async version should be used. So default (0) stream should also return true.

Reply:
That was my motivation for having this check - should a comment be added to indicate the reason for it? (I realise now it may not be obvious from looking at this test in isolation without knowing how async / non-async are chosen)

Reply:
I've decided to add the comment anyway, to save anyone wondering about it in future having to dig through the history and find this PR discussion to understand what's going on there :-)